### PR TITLE
add shard size to dataframe_to_xshards

### PR DESCRIPTION
**Pull request by yushan111**

### CODE REVIEW

**Comment by yushan111** [comment]
Interface change:
Add `shard_size` in `fit`, `predict` and `evaluate` of `PyTorchRayEstimator` and `TensorFlow2Estimator`. 
`shard_size` is for Spark DataFrame input, which specifies the number of Rows to transform as one shard of SparkXShards. It defaults to None, in which case Rows in one partition will be transformed as one shard.

**Comment by jason-dai** [comment]
> Interface change:
> Add `shard_size` in `fit`, `predict` and `evaluate` of `PyTorchRayEstimator` and `TensorFlow2Estimator`.
> `shard_size` is for Spark DataFrame input, which specifies the number of Rows to transform as one shard of SparkXShards. It defaults to None, in which case Rows in one partition will be transformed as one shard.

I don't think it's reasonable to expose `shard_size` to the user; we can add an internal property in `OrcaContext` for test purpose.

File: pyzoo/test/zoo/orca/learn/ray/tf/test_tf_ray_estimator.py [hunk 1]
```diff
@@ -347,6 +347,33 @@ def test_dataframe(self):
                          label_cols=["label"])
         trainer.predict(df, feature_cols=["feature"]).collect()
 
+    def test_dataframe_shard_size(self):
+        from zoo.orca import OrcaContext
+        OrcaContext.shard_size = 3
```
Comment (yangw1234) [inline pyzoo/test/zoo/orca/learn/ray/tf/test_tf_ray_estimator.py:352]: add a test to check whether this takes effect?

**Comment by yangw1234** [comment]
LGTM

**Comment by yushan111** [comment]
> I don't think it's reasonable to expose `shard_size` to the user; we can add an internal property in `OrcaContext` for test purpose.

`shard_size` has been added in `OrcaContext`

File: pyzoo/zoo/orca/common.py [hunk 2]
```diff
@@ -101,6 +102,23 @@ def train_data_store(cls, value):
             "train_data_store must be either DRAM or PMEM or DIRECT or DISK_n"
         cls._train_data_store = value
 
+    @property
+    def shard_size(cls):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/common.py:106]: `_shard_size`, as it should be used for test only.

**Comment by yushan111** [comment]
Jenkins:
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/5106/
http://10.239.47.210:18888/job/ZOO-PR-Python-Spark-2.4-py36-ray/1321/

**Pull request merged**